### PR TITLE
Make checking in and publishing code simpler/more correct

### PR DESCRIPTION
**Pull request by phil303**
tl;dr pretty similar to https://github.com/reddit/node-api-client/pull/157 but the details are a little different due to the structure of the library.

Problem:
Currently we check-in our transpiled code. Other than this being a
general nuisance, it's actually a little dangerous as well. Because it's
essentially one big blob of code, it's hard to read and parse. We've
already encountered situations when someone's checked in code bashed
someone else's changes.

Fix:
It's fairly simple to fix this by checking in only the source code and
publishing only the build code.

Details:
Due to the way platform exports its modules as default exports, we need
to gitignore all of their transpiled modules. We do the opposite for
npmignore, ignoring mostly everything but the transpiled modules.

Then update package.json to have a prepublish hook to ensure that our
source code is transpiled before it's pushed to npm.

:eyeglasses: @schwers || @nramadas || @griffinmichl

### CODE REVIEW

**Comment by phil303** [comment]
:haircut: @schwers

**Comment by schwers-zz** [comment]
:fish:
